src/pybind/mgr/cephadm/module.py

```diff
@@ -2886,6 +2886,11 @@ def get_daemon_names(daemons: List[str]) -> List[str]:
             assert ingress_spec.backend_service
             daemons = self.cache.get_daemons_by_service(ingress_spec.backend_service)
             deps = [d.name() for d in daemons]
+            for attr in ['ssl_cert', 'ssl_key']:
+                ssl_cert_key = getattr(ingress_spec, attr, None)
+                if ssl_cert_key:
+                    assert isinstance(ssl_cert_key, str)
+                    deps.append(ssl_cert_key)
```

Comment:
Just as a suggestion: `ssl_cert` and `ssl_key` are big strings ... maybe it's good enough to just add some hash of them as deps instead of pushing the certs themselves. That could lead to a messy output.

Reply:
made this change